src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,236 @@
+        )
+
+    meas_list = [gates.Z, gates.X, gates.Y, gates.Z]
+    j_to_measurements = list(product(meas_list, repeat=nqubits))[j]
+    new_circ = circ.copy()
+    for q in range(len(j_to_measurements)):
+        meas_q = j_to_measurements[q]
+        new_circ.add(gates.M(q, basis=meas_q))
+
+    return new_circ
+
+
+def reset_register(circuit, invert_register):
+    """Returns an inverse circuit of the selected register.
+        One can then add inverse_circuit to the original circuit by addition:
+            circ_with_inverse = circ.copy()
+            circ_with_inverse.add(inverse_circuit.on_qubits(invert_register))
+        where register_to_reset = 0, 1, or [0,1].
+
+        Args:
+        circuit (:class:`qibo.models.Circuit`): original circuit
+        invert_register (string): Qubit(s) to reset:
+            'sp_0' (qubit 0);
+            'sp_1' (qubit 1); or
+            'sp_t' (both qubits)
+    Returns:
+        inverse_circuit (:class:`qibo.models.Circuit`): Inverse of the input circuit's register.
+    """
+
+    if invert_register == "sp_0" or invert_register == "sp_1":
+        if invert_register == "sp_0":
+            register_to_reset = 0
+        elif invert_register == "sp_1":
+            register_to_reset = 1
+
+        new_circ = Circuit(1)
+        for data in circuit.raw["queue"]:
+            theta = data.get("init_kwargs", {}).get("theta", None)
+            if data["_target_qubits"][0] == register_to_reset:
+                if theta is None:
+                    new_circ.add(getattr(gates, data["_class"])(0))
+                else:
+                    new_circ.add(
+                        getattr(gates, data["_class"])(0, theta)
+                    )  # <-- Not covered in tests. Thinking about future proofing when we need to invert a random gate that has theta parameter.
+
+    elif invert_register == "sp_t":
+        new_circ = circuit.copy()
+
+    else:
+        raise NameError(
+            f"{invert_register} not recognized. Input "
+            "sp_0"
+            " to reset qubit 0, "
+            "sp_1"
+            " to reset qubit 1, or "
+            "sp_t"
+            " to reset both qubits."
+        )
+
+    inverse_circuit = new_circ.invert()
+
+    return inverse_circuit
+
+
+def GST_execute_circuit(circuit, k, j, nshots=int(1e4), backend=None):
+    """Executes a circuit used in gate set tomography and processes the
+        measurement outcomes for the Pauli Transfer Matrix notation. The circuit
+        should already have noise models implemented, if any, prior to using this
+        function.
+
+        Args:
+        circuit (:class:`qibo.models.Circuit`): The Qibo circuit to be executed.
+        k (int): The index of the state prepared.
+        j (int): The index of the measurement basis.
+        nshots (int, optional): Number of shots to execute circuit with.
+    Returns:
+        numpy.float: Expectation value given by either Tr(Q_j rho_k) or Tr(Q_j O_l rho_k).
```

Comment:
I would suggest to also introduce this notation in the short overview of GST in the docs.